### ISSUE TITLE
file:// links not affected

### ISSUE THREAD

**Issue by jkopczyn** [darkreader/darkreader#2274]
To reproduce: save an arbitrary text file to disk locally, then open it in Chrome. (Probably affects other browsers as well.) Toggle the extension; it will have no effect.

Suggested fix: Unclear. The obvious solution would be CSS that applies to everything, which I imagine would negatively impact the ability to exclude elements from the swap on any other page. I don't think there's a CSS selector for "this has no class", which might be more workable if it does exist.

**Comment by Gusted** [comment]
The extension doesn't work for local files
Open chrome://extensions page, find Dark Reader, click Allow access to file URLs

**Comment by jkopczyn** [comment]
> The extension doesn't work for local files

Yes, that is the bug I am reporting here. This is a bug which gets in the way of the extension working properly.

**Comment by FranklinYu** [comment]
This extension does work for local files as long as you enable it in extension settings.

**Comment by Gusted** [comment]
> > The extension doesn't work for local files
> 
> Yes, that is the bug I am reporting here. This is a bug which gets in the way of the extension working properly.

You need to enable it in chrome://extensions security measures from the browser, but last time I checked it's already toggled. I'm not seeing any issues here or I'm misunderstanding you.

**Comment by jkopczyn** [comment]
File links are enabled in extension settings. It still does not function on file links.

**Comment by Gusted** [comment]
Which version of Chromium does your browser use?
It's working for me fine https://i.imgur.com/x3FCtPe.gifv

**Comment by FranklinYu** [comment]
I also failed to reproduce this bug on macOS with latest Chrome.

**Comment by unai-ndz** [comment]
I can reproduce on Firefox, but only the Dynamic mode fails.

**Comment by Gusted** [comment]
Firefox doesn't have any support for it so far I know

**Comment by unai-ndz** [comment]
It works just fine for almost every page, as long as it's not a local file.

**Comment by Gusted** [comment]
file:// is local file 🤦 

> It works just fine for almost every page, as long as it's not a local file.

**Comment by unai-ndz** [comment]
Ohh I thought you were referring to Dynamic mode in general, sorry about that.

**Comment by jkopczyn** [comment]
Chrome 81.0.4044.129 (For Windows, ChromeOS is reliably on newest stable build but I don't have it right here.)
Consistently recurs on both Windows and ChromeOS.
Don't see any options for "Dynamic mode" or any other mode. Tried forcing an update to the extension, still don't see it. Definitely have it allowed on file:// links.

That option doesn't show up for me. I tried updating the extension and it still does not show up for me. Chrome 81.0.4044.129. Consistently recurs on both Windows and ChromeOS.

**Comment by FranklinYu** [comment]
FYI I failed to reproduce on Chrome OS. Don’t have Windows to test it.

**Comment by illucent** [comment]
Feature request. Similar issue (with online files also) and I propose different approach: to add regex or wildcard support  to "Site list" for files ending with e.g. "*.pdf". / by file extension.
```
Chromium | 81.0.4044.138 (RPM Fusion Build) (64-bit)
Revision | 8c6c7ba89cc9453625af54f11fd83179e23450fa-refs/branch-heads/4044@{#999}
OS | Linux
```
![20200512152132EEST2020_ch5pdfChromium](https://user-images.githubusercontent.com/3680424/81691181-1699b000-9465-11ea-8640-cb187a30a30c.png)

**Comment by Gusted** [comment]
> Feature request. Similar issue (with online files also) and I propose different approach: to add regex or wildcard support to "Site list" for files ending with e.g. "*.pdf". / by file extension.
> 
> ```
> Chrome | 81.0.4044.138 (RPM Fusion Build) (64-bit)
> Revision | 8c6c7ba89cc9453625af54f11fd83179e23450fa-refs/branch-heads/4044@{#999}
> OS | Linux
> ```
> 
> ![20200512152132EEST2020_ch5pdfChromium](https://user-images.githubusercontent.com/3680424/81691181-1699b000-9465-11ea-8640-cb187a30a30c.png)

We don't support *.pdf anymore as that doesn't fit well in our current system.

For pdf's however you can toggle it

Dev tools - > preview new design
Settings - > pdf

**Comment by illucent** [comment]
Thanks for reply, any document file etension *.docx, *.md, etc.

**Comment by Gusted** [comment]
> Thanks for reply, any document file extension *.docx, *.md, etc.

^.docx
^.md

**Comment by MostHated** [comment]
For local files (the particular one I am wanting to tweak is file:///C:/Program%20Files/Wexflow/Backend/designer.html) in the devtools dynamic theme editor, what is to be put in for the url? It works as is, but there are some specific elements I need to adjust.

**Comment by Gusted** [comment]
![](https://i.imgur.com/aipPU2Y.png) Not the best option but make sure that you use a good selector.

**Comment by renfenghuan** [comment]
I think it doesn't work for some specific files. Like the one I downloaded from nexus:

https://www.nexusmods.com/skyrimspecialedition/mods/24673?tab=posts

This is a local subset version of a website.

I've already had the "Allow Access to File URLs" on.

I can successfully see the scroll bar goes dark. But the rest part of the page remain white.

**Comment by FranklinYu** [comment]
@renfenghuan I tried the “Download: Manual” button. It requires an account to download, so I can’t reproduce what you said.

**Comment by renfenghuan** [comment]
> @renfenghuan I tried the “Download: Manual” button. It requires an account to download, so I can’t reproduce what you said.

Just register one, ok? it's not a big deal.

**Comment by Gusted** [comment]
@renfenghuan Open up a new issue, thank you. Don't want any further discussion on a Issue that isn't related
